scripts/ts-bin/pair_comments_for_relativerating.ts

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+/*
+Usage: Convert answers with multiple parts into separate rows.
+
+  ts-node ./ts-bin/pair_comments_for_relativerating.ts \
+    --infile="./tmp/path-to-input.jsonl" \
+    --output_path_prefix="./tmp/path-to-paired"
+*/
+
+import * as yargs from 'yargs';
+import * as fs from 'fs';
+import * as readline from 'readline';
+// import * as jsonexport from "jsonexport/dist"
+// import * as stream from 'stream';
+
+// Command line arguments.
+interface Params {
+  infile:string,
+  output_path_prefix:string,
+  output_path_postfix:string,
+  left_id_prefix:string,
+  right_id_prefix:string,
+  left_text_prefix:string,
+  right_text_prefix:string,
+  entries_per_row:number,
+  rows_per_file:number,
+};
+
+interface InputObj {
+  comment_text:string;
+  id:string;
+}
+
+// TODO(ldixon): rewrite based on observables, and its sensible abstractions
+// of streams. This will also enable this to scale to more data than fits in
+// memory.
+async function main(args : Params) {
+    let instream = fs.createReadStream(args.infile);
+    let rl = readline.createInterface(instream);
+
+    let lineCount = 0;
+
+    let data : InputObj[] = [];
+    let pair_id = 1;
+    // let pairs_to_ouput : { [key:string]:string }[] = [];
+    let pair_id_postfix = '';
+    let out_pairs_for_row: { [key:string]:string } = {};
+    let out_rows_for_file: { [key:string]:string }[] = [];
+    let out_files: { [key:string]:string }[][] = [];
+
+    rl.on('line', async function(line) {
+      let obj : InputObj = JSON.parse(line);
+      lineCount++;
+      data.push(obj);
+    });
+
+    // instream.pipe(jsonexport()).pipe(writer);
+    // var writer = fs.createWriteStream('out.csv');
+
+    rl.on('close', function() {
+
+      for(let d1 of data) {
+        for(let d2 of data){
+          if (args.entries_per_row > 1) {
+            pair_id_postfix = `${pair_id}`
+          }
+
+          out_pairs_for_row[`a.p.file_id`] = `${out_files.length + 1}`;
+          out_pairs_for_row[`a.p.row_id`] = `${out_rows_for_file.length + 1}`;
+          out_pairs_for_row[`${args.left_id_prefix}${pair_id_postfix}`] = d1.id;
```

Comment:
I'm a bit confused by this. For example, why do we put the ids for the row and the comment text both in the same dictionary? Is there some sort of parsing done on the other end to extract them? It would be helpful to have some documentation explaining the logic here, or some functions with more descriptive names to populate these fields.

Reply:
For the followup tool. We don't get a lot of choice.